lms/djangoapps/courseware/utils.py

```diff
@@ -32,17 +35,35 @@ def verified_upgrade_deadline_link(user, course=None, course_id=None):
     return EcommerceService().upgrade_url(user, course_id)
 
 
-def verified_upgrade_link_is_valid(enrollment=None):
+def can_show_verified_upgrade(user, enrollment, course=None):
     """
-    Return whether this enrollment can be upgraded.
+    Return whether this user can be shown upgrade message.
 
     Arguments:
+        user (:class:`.AuthUser`): The user from the request.user property
         enrollment (:class:`.CourseEnrollment`): The enrollment under consideration.
             If None, then the enrollment is considered to be upgradeable.
+        course (:class:`.ModulestoreCourse`): Optional passed in modulestore course.
+            This is the course object to retrieve partitions from
```

Comment:
Am I right that this is essentially an optimization, allowing a pre-loaded to course (that matches `enrollment.course.id`) to be passed in?

If so, I suggest clarifying that in the docstring:
```suggestion
        course (:class:`.ModulestoreCourse`): Optional passed in modulestore course.
            If provided, it is expected to correspond to `enrollment.course.id`.
            If not provided, the course will be loaded from the modulestore.
            We use the course to retrieve user partitions when calculating whether
            the upgrade link will be shown.            
```

Reply:
Correct. If I don't get the course objects passed in, we call module store to load course twice which is a performance penalty.